### PR TITLE
Fixed #20488 -- Added copy and move methods to the storage API

### PR DESCRIPTION
**Pull request by kux**
* implemented these methods in the default FileSystemStorage
* fix issue with storage tests: each test from FileStorageTests
   was run twice (when inheriting from a TestCase class all test_
   methods are ran in the subclass as well)

### CODE REVIEW

**Comment by kux** [comment]
buildbot, test this please

File: docs/ref/files/storage.txt [hunk 1]
```diff
@@ -95,6 +95,15 @@ The Storage Class
...
         ``NotImplementedError`` instead.
 
+    .. method:: copy(src_name, dst_name)
+
+        Copy the file specified by ``src_name`` to ``dst_name``. If
+        ``dst_name`` already exists ``ValueError`` will be raised. For
+        storage systems that don't support copy this will raise
+        ``NotImplementedError`` instead.
+
+        .. versionadded:: 1.8
```
Comment (collinanderson) [inline docs/ref/files/storage.txt]: 1.9 at this point.

**Comment by collinanderson** [comment]
buildbot, test this please

File: docs/ref/files/storage.txt (hunk 1 above)
Comment (kux) [inline docs/ref/files/storage.txt]: Is it? I don't see 1.8 being released yet.

Comment (berkerpeksag) [inline docs/ref/files/storage.txt]: master is now in feature freeze mode. See https://code.djangoproject.com/wiki/Version1.8Roadmap for the schedule and release process.

Comment (kux) [inline docs/ref/files/storage.txt]: Well, this means that I also need to move the changes to the release notes in an 1.9.txt

However, that file doesn't exist yet. Is there any template for creating release notes for a new version?

Comment (MarkusH) [inline docs/ref/files/storage.txt]: The template will be added one we create the stable/1.8.x branch.

Comment (collinanderson) [inline docs/ref/files/storage.txt]: (which will likely happen in the next few days if not tomorrow)

Comment (timgraham) [inline docs/ref/files/storage.txt]: Ioan, 1.9 release notes are there now if you have a chance to update this.

**Comment by kux** [comment]
Rebased and opened https://github.com/django/django/pull/4244 with release notes in 1.9.txt rather than 1.8.txt

File: docs/ref/files/storage.txt (hunk 1 above)
Comment (kux) [inline docs/ref/files/storage.txt]: Rebased and opened #4244 with release notes in 1.9.txt rather than 1.8.txt

**Pull request closed without merging**